src/Workspaces/Core/Portable/Options/EditorConfigStorageLocation.cs

```diff
@@ -49,11 +53,19 @@ public EditorConfigStorageLocation(string keyName)
             {
                 if (type == typeof(int))
                 {
-                    return int.Parse(s);
+                    if (int.TryParse(s, out var value))
+                    {
+                        return value;
+                    }
+                    return null;
                 }
                 else if (type == typeof(bool))
                 {
-                    return bool.Parse(s);
+                    if (bool.TryParse(s, out var value))
+                    {
+                        return value;
+                    }
+                    return null;
```

Comment:
null in this case means use the default value.

---
In reply to: [102832596](https://github.com/dotnet/roslyn/pull/17097#discussion_r102832596) [](ancestors = 102832596)